firefox-ios/Client/Frontend/Browser/BrowserViewController/State/BrowserViewControllerState.swift

```diff
@@ -256,6 +257,18 @@ struct BrowserViewControllerState: ScreenState, Equatable {
                 navigateTo: .forward,
                 showQRcodeReader: false,
                 microsurveyState: MicrosurveyPromptState.reducer(state.microsurveyState, action))
+        case GeneralBrowserActionType.showBackForwardList:
+            return BrowserViewControllerState(
+                searchScreenState: state.searchScreenState,
+                showDataClearanceFlow: state.showDataClearanceFlow,
+                toolbarState: state.toolbarState,
```

Comment:
explicitly call the reducer here?

Reply:
Could you please elaborate on the benefits of doing so in this context? I'd like to understand how this change would improve the state management.

Reply:
ah this is related to my comment here.
https://github.com/mozilla-mobile/firefox-ios/pull/20428#discussion_r1624378049

so I was proposing using `ToolBarState.reducer()` instead of `state.toolbarState`